js/interface.js

```diff
@@ -1243,6 +1243,13 @@ var DynamicLists = (function() {
         }
       });
     },
+    getRowColumnValue: function(item) {
```

Comment:
I don't know what `getRowColumnValue` means.

Reply:
I added description comment, this name is based on function usages

Reply:
Would `getSelectedColumnValue` be better for comprehension?

Reply:
It sounds like the scenario is that you have an `item`, which contains `item.column`. You're validating `item.column` to make sure it's part of `item.columns`.

Why are `item.column === 'empty'` and `item.column === 'custom'` exceptions?

Reply:
Yes, because when we set field value with `empty` or `custom` this value is written in `item.column` and `item.columns` won't include it.

Reply:
Why and where is `empty` and `custom` saved into `item.columns`?

Reply:
They are saved in `'item.column`, `item.columns` is an array of data source columns or default columns.
When we PUT widget-instance, the `summary-fields` looks like:
![image](https://user-images.githubusercontent.com/74735207/102892081-04a2d580-4468-11eb-9db5-0cf633138635.png)

When we render `Data view options` we have next scenarios for setting selected summary fields' values:

1. Data source or default columns include `item.column` -> show column name
2. Data source or default column doesn't include `item.column` -> show `-- Select a data field`
3. `item.column` is `none` -> show `-- Select a data field`
4. `item.column` is `custom` and `customFieldEnabled === true` -> show `Custom` and custom field
5. `item.column` is `empty` -> show 'None'

Same scenarios are applied to detail-fields
This function `getRowColumnValue` returns `item.column` or `null` to show correct option value in select
At rows 1208 and 1230 we set selected field value as `item.column || 'none'`
If you approve I will change return of this function from `null` to `'none'` and at rows 1208 and 1230 change `item.column || 'none'` to `item.column`.

Reply:
@ivandevupp Thanks for explaining. I get it now. In this case, I'd name it slightly different with a different purpose.

```suggestion
    /**
     * Validates a column name from field settings
     */
    validateColumn: function(options) {
```